src/coreclr/jit/lowerxarch.cpp

```diff
@@ -3419,45 +3419,73 @@ GenTree* Lowering::LowerHWIntrinsicDot(GenTreeHWIntrinsic* node)
         {
             assert(simdBaseType == TYP_FLOAT);
 
+            // We need to mask off the most significant element to avoid the shuffle + add
+            // from including it in the computed result. We need to do this for both op1 and
+            // op2 in case one of them is `NaN` (because Zero * NaN == NaN)
+
+            simd12_t simd12Val = {};
```

Comment:
This depends on simd12 constants getting zero-extended once they get loaded into register. It is not guaranteed to happen. For example, https://github.com/dotnet/runtime/blob/04690205c07a07a242155e88fff0acb730f3dc73/src/coreclr/jit/codegenarm64.cpp#L2402-L2405 is not going to zero extend them.

Should we rather use simd16 constant here?

Reply:
Ah right, `IsAllBitsSet` is specially handled so https://github.com/dotnet/runtime/pull/74857 won't cover it.

Will adjust it.